### PR TITLE
[FIX] IDPEP fails on certain datasets. Fixes #1376.

### LINKED ISSUE #1376: IDPosteriorErrorProbability does not finish
The IDPosteriorErrorProbability tool runs for >12 hours on this file [1] but does not finish. Tested on Windows and OS X. Neither warnings or progress are given. The toppas pipeline [1] shows the previous steps of the workflow.

[1] https://www.dropbox.com/sh/21jrrokl3rl2tjr/AABuqUt5BiMbhxuSh3LOucpSa

### PR DESCRIPTION
**Pull request by jpfeuffer**
Introduces parameter to manually set the maximum number of iterations and removes the smallest_e_value parameter which is set to the smallest possible value by default.
Fixes #1376.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Nice thanks to you both. Should we consider that minimal decreases in likelihood might happen?

**Comment by timosachsenberg** [comment]
hi there are some test failing

**Comment by lars20070** [comment]
@jpfeuffer Could you please fix the tests? I will then update our server installation. Thanks.

**Comment by lars20070** [comment]
Sorry. IDPEP tests still fail.

**Comment by jpfeuffer** [comment]
Yup, unfortunately. Working on it. It's only the style check and the class test left.

**Comment by jpfeuffer** [comment]
OK some tests with GUI are randomly failing. I have no idea. But style check, tests on my MacOS and the clang tests without GUI are passing now.

**Comment by timosachsenberg** [comment]
Error: File not found (the file 'IDPosteriorErrorProbability_bad_data_output.tmp' could not be found)
Cannot read input file given from parameter '-in1'!
Strange that this one doesn't run through. Otherwise it looks good

**Comment by lars20070** [comment]
I restarted the build. Apart from the style check, now all three tests fail with

1599 - TOPP_IDPosteriorErrorProbability_7_out1 (Failed)

**Comment by jpfeuffer** [comment]
I see. So, the thing is, this is not my test. In the code, there was a question why there is no test for this, so I included one. Now I see, why they removed it.
It checks the -ignore_bad_data option and it seems that on my machine it runs and produces output, but on other platforms it does not (i.e. not producing the .tmp output).
Options:
- Could you test
    ${TOPP_BIN_PATH}/IDPosteriorErrorProbability -test -in ${DATA_DIR_TOPP}/IDPosteriorErrorProbability_bad_data.idXML -out IDPosteriorErrorProbability_bad_data_output.tmp -ignore_bad_data
on your machines quickly?
- Ignore and remove the test again. Or remove the option completely. It might be handy for loops in pipelines but also dangerous, since it seems to do unpredictable things.

**Comment by lars20070** [comment]
@jpfeuffer I checked out your branch on my Mac. Two tests are failing.

348 - ConsensusIDAlgorithmBest_test (Failed)
1564 - TOPP_ConsensusID_5_out1 (Failed)

**Comment by jpfeuffer** [comment]
Thank you very much. That speaks for a compiler/platform specific problem with the -ignore_bad_data option. ConsensusID has nothing to do with my changes and it doesnt fail on my MAC. (ALthough IDPEP and ConsensusID are related, the input files are fixed for the ConsensusID tests).
I will just remove the test for this option (like before) and add a "(not well tested)" to the docu of this option.

**Comment by lars20070** [comment]
@hendrikweisser Can you comment in this? And #1427.

**Comment by timosachsenberg** [comment]
+1

File: src/openms/source/MATH/STATISTICS/PosteriorErrorProbabilityModel.cpp [hunk 1]
```diff
@@ -190,13 +195,18 @@ namespace OpenMS
...
 
         double new_maxlike(computeMaxLikelihood(incorrect_density, correct_density));
-        if (boost::math::isnan(new_maxlike - maxlike))
+        if (boost::math::isnan(new_maxlike - maxlike) || new_maxlike < maxlike)
         {
           return false;
           //throw Exception::UnableToFit(__FILE__,__LINE__,__PRETTY_FUNCTION__,"UnableToFit-PosteriorErrorProbability","Could not fit mixture model to data");
         }
-        if (fabs(new_maxlike - maxlike) < 0.001)
+        if ((new_maxlike - maxlike) < pow(10,-delta) || itns >= max_itns)
```
Comment (hendrikweisser) [inline src/openms/source/MATH/STATISTICS/PosteriorErrorProbabilityModel.cpp]: Compilation error on Mac with clang:
```
/Users/hw5/Software/OpenMS/src/openms/source/MATH/STATISTICS/PosteriorErrorProbabilityModel.cpp:203:39: error: call to 'pow' is ambiguous
        if ((new_maxlike - maxlike) < pow(10,-delta) || itns >= max_itns)
                                      ^~~
/usr/include/c++/4.2.1/cmath:364:3: note: candidate function
  pow(double __x, int __i)
  ^
/usr/include/c++/4.2.1/cmath:368:3: note: candidate function
  pow(float __x, int __n)
  ^
/usr/include/c++/4.2.1/cmath:372:3: note: candidate function
  pow(long double __x, int __n)
  ^
/usr/include/math.h:436:15: note: candidate function
extern double pow(double, double);
              ^
/usr/include/c++/4.2.1/cmath:356:3: note: candidate function
  pow(float __x, float __y)
  ^
/usr/include/c++/4.2.1/cmath:360:3: note: candidate function
  pow(long double __x, long double __y)
  ^
```

Comment (hendrikweisser) [inline src/openms/source/MATH/STATISTICS/PosteriorErrorProbabilityModel.cpp]: Changing "10" to "10.0" fixes it.

**Comment by timosachsenberg** [comment]
thanks - can you open a PR for a quick fix?

**Comment by hendrikweisser** [comment]
> thanks - can you open a PR for a quick fix?

Done: https://github.com/OpenMS/OpenMS/pull/1438

**Pull request merged**